### ISSUE TITLE
ProtocolError: malformed MIME header initial line

### ISSUE THREAD

**Issue by reggi** [denoland/deno_std#40]
I'm sending over a `POST` request and I think it's confusing the body for the header.

`ProtocolError: malformed MIME header initial line:   "query": "{ hello }",`

**Comment by kevinkassimo** [comment]
Could you post the raw response that triggers this error?

**Comment by reggi** [comment]
```
{
  "query": "{ hello }",
  "operationName": "",
  "variables": {}
}
```

**Comment by kevinkassimo** [comment]
Actually, have you sent the request correctly? Because it seems as if it is treating `"query": "{ hello }",` as the initial header line, which should only happen when it reads the first line `{` as the request line... (looks as if no request line nor headers are sent along with the body...)

**Comment by bartlomieju** [comment]
I managed to reproduce this issue, and it turns out this problem arises if you try to respond to request without reading its body.

`http_bench.ts`
```ts
import * as deno from "deno";
import { serve } from "./http.ts";

const addr = deno.args[1] || "127.0.0.1:4500";
const server = serve(addr);

const body = new TextEncoder().encode("Hello World");

async function main(): Promise<void> {
  for await (const request of server) {
    await request.body() // adding this line prevent ProtocolError
    await request.respond({ status: 200, body });
  }
}

main();
``` 

I guess a check is needed in `request.respond` that ensures whole requests has been read.

**Comment by bartlomieju** [comment]
I believe this was resolved in #49, can we close this now?
